ethereum_bridge/src/storage/vote_tallies.rs

```diff
@@ -107,6 +119,70 @@ impl From<&Hash> for Keys<EthereumEvent> {
     }
 }
 
+/// A wrapper struct for managing keys related to
+/// tracking signatures over bridge pool roots.
+#[derive(Clone)]
+pub struct BridgePoolRoot(pub KeccakHash);
+
+impl BorshSerialize for BridgePoolRoot {
+    fn serialize<W: Write>(&self, writer: &mut W) -> std::io::Result<()> {
+        BorshSerialize::serialize(&self.0, writer)
+    }
+}
+
+impl BorshDeserialize for BridgePoolRoot {
+    fn deserialize(buf: &mut &[u8]) -> std::io::Result<Self> {
+        <KeccakHash as BorshDeserialize>::deserialize(buf).map(BridgePoolRoot)
+    }
+}
+
+/// A wrapper struct for managing keys related to
+/// tracking signatures over bridge pool nonces.
+#[derive(Clone)]
+pub struct BridgePoolNonce(pub Uint);
+
+impl BorshSerialize for BridgePoolNonce {
+    fn serialize<W: Write>(&self, writer: &mut W) -> std::io::Result<()> {
+        BorshSerialize::serialize(&self.0, writer)
+    }
+}
+
+impl BorshDeserialize for BridgePoolNonce {
+    fn deserialize(buf: &mut &[u8]) -> std::io::Result<Self> {
+        <Uint as BorshDeserialize>::deserialize(buf).map(BridgePoolNonce)
+    }
+}
+
+impl From<BridgePoolRoot> for Keys<BridgePoolRoot> {
+    fn from(hash: BridgePoolRoot) -> Self {
+        let hash = hash.0.to_string();
+        let prefix = super::prefix()
+            .push(&BRIDGE_POOL_ROOT_PREFIX_KEY_SEGMENT.to_owned())
+            .expect("should always be able to construct this key")
+            .push(&hash)
+            .expect("should always be able to construct this key");
+        Keys {
+            prefix,
+            _phantom: std::marker::PhantomData,
+        }
+    }
+}
+
+impl From<BridgePoolNonce> for Keys<BridgePoolNonce> {
+    fn from(nonce: BridgePoolNonce) -> Self {
+        let nonce = nonce.0.to_string();
+        let prefix = super::prefix()
+            .push(&BRIDGE_POOL_NONCE_PREFIX_KEY_SEGMENT.to_owned())
+            .expect("should always be able to construct this key")
+            .push(&nonce)
+            .expect("should always be able to construct this key");
+        Keys {
+            prefix,
+            _phantom: std::marker::PhantomData,
+        }
+    }
+}
+
```

Comment:
this should be moved elsewhere. maybe `ethereum_bridge/src/storage/bp_roots.rs`?